### ISSUE TITLE
Critical: Permanent FC immediately after startup with Search history problem

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#1853]
Had one customer report this today: OsmAnd dies on him  with an FC immediately after startup, see log attached. Last activities had been adding a favorite, doing some searches and downloading some new maps. OsmAnd continued to function ok until the next app star, then thishappendt. Now there seems no way to make it functional again.

I assume that deleting the app data in the Android application settings would reset this for him and will let him try unless anybody has a better idea or we can fix this quickly in the code.

Looks like a critical error we need to amend asap. (Going back to older builds until August 8 did not help)

Version  OsmAnd~ 2.2.0#13910M
30.10.2015 4:57:43
Apk Version : 2.2.0#13910M 211
Exception occured in thread Thread[main,5,main] : 
java.lang.RuntimeException: Unable to resume activity {net.osmand.plus/net.osmand.plus.activities.MapActivity}: java.lang.NullPointerException
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3052)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3093)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396)
	at android.app.ActivityThread.access$600(ActivityThread.java:151)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1335)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:155)
	at android.app.ActivityThread.main(ActivityThread.java:5520)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:511)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1029)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:796)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
	at net.osmand.plus.activities.search.SearchHistoryFragment.udpateHistoryItem(SearchHistoryFragment.java:271)
	at net.osmand.plus.dashboard.DashRecentsFragment.setupRecents(DashRecentsFragment.java:96)
	at net.osmand.plus.dashboard.DashRecentsFragment.onOpenDash(DashRecentsFragment.java:72)
	at net.osmand.plus.dashboard.DashBaseFragment.onResume(DashBaseFragment.java:123)
	at android.support.v4.app.Fragment.performResume(Fragment.java:1832)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:995)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1138)
	at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:740)
	at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1501)
	at android.support.v4.app.FragmentActivity.onPostResume(FragmentActivity.java:437)
	at android.support.v7.app.AppCompatActivity.onPostResume(AppCompatActivity.java:138)
	at android.app.Activity.performResume(Activity.java:5161)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3035)
	... 12 more

**Comment by sonora** [comment]
Looks like it is nothing we introduced recently. I installed an old build of March 1 on that device and then the exception log reports this line to fail: https://github.com/osmandapp/Osmand/blob/494b0c2652dfaa89ab1f144cd34399dcf0772522/OsmAnd/src/net/osmand/plus/activities/search/SearchHistoryFragment.java#L184, which is the very same line reported in exception logs for current builds: https://github.com/osmandapp/Osmand/blob/master/OsmAnd/src/net/osmand/plus/activities/search/SearchHistoryFragment.java#L271. 

So it looks like somehow there is a way for the Search history to become "corrupt", and then OsmAnd will not start up anymore, making OsmAnd completely unusable. Scary, we need to investigate. As expected, deleting OsmAnd's data in the Android app settings, which essentially resets OsmAnd to being freshly installed, resolved the issue.

Looks to me like the issue came up by the user having an entry in their search history which (or the name of which) subsequently became "unrecognizable", either by a map or wiki update, or by renaming/deleting a Favorite, etc.
